scripts/system_performance/run_statistics.py

```diff
@@ -0,0 +1,292 @@
+        method_mappings = {'missing_run_numbers_report': self.missing_run_numbers_report,
+                           'execution_time': self.execution_time,
+                           'execution_time_average': self.execution_time_average,
+                           'run_frequency': self.run_frequency,
+                           'run_frequency_average': self.run_frequency_average
+                           }
+        return method_mappings
+
+    @staticmethod
+    def list_all_instruments():
+        """Returns a list of all currently active instruments"""
+        instrument_list = DatabaseMonitorChecks.list_instruments()
+        return instrument_list
+
+    def run_every_instruments(self, instrument_dict, method_name):
+        """Returns all existing instruments in a dictionary for a given method as:
+        {instrument_name : [[method output row 1]],[[method output row 2] etc] ...}"""
+        for sublist in self.list_all_instruments():
+            try:
+                if method_name in self.create_method_mappings():
+                    instrument_dict[sublist[1]] = self.create_method_mappings()[method_name](sublist[0])
+            except KeyError:
+                raise ValueError('Invalid Input - method {} does not exist'.format(method_name))
+        return instrument_dict
+
+    def input_verification(self, instrument_input, method_name):
+        """Checks that the instrument's in method input exist and then calls
+        and returns the methods used as input for each instrument placing in
+        a dictionary as a nested list for each instrument as:
+         {instrument_name : [[method output row 1]],[[method output row 2] etc.] ...}"""
+        instrument_dict = {}
+
+        for sublist in self.list_all_instruments():
+            # if input is None, run all instruments by default
+            if not isinstance(instrument_input, list):
+                logging.error("Value is not iterable, the first argument must be of type list")
+                return None
+            for instrument in instrument_input:
+                if sublist[1] == instrument:
+                    # check input is in mapping and place method output in instrument_dict
+                    try:
+                        if method_name in self.create_method_mappings():
+                            instrument_dict[sublist[1]] = self.create_method_mappings()[method_name](sublist[0])
+                    except KeyError:
+                        raise ValueError('Invalid Input - method {} does not exist'.format(method_name))
+                # Run all methods is user specifies "all"
+                elif instrument == 'all' or not instrument:
+                    self.run_every_instruments(instrument_dict, method_name)
+                else:
+                    logging.info('The instrument: {} is not currently using Autoreduction'.format(instrument))
```

Comment:
`The instrument: {} has not been found in the autoreduction database`
The above is more obvious what the error is for developers (which is the main audience for this error)